Prevent uncaught exception in some cases

wwwouter opened the pull request:
If something goes wrong in the code contained by setImmediate, an uncaughtException is thrown.

The code in this pull requests wraps the code in a try/catch block and emits an error if an error is thrown.

kevva commented:
Could you add a test for this?

sindresorhus commented on index.js:
```diff
@@ -146,55 +146,59 @@ function requestAsEventEmitter(opts) {
-				const progressStream = new Transform({
-					transform(chunk, encoding, callback) {
-						downloaded += chunk.length;
+				try {
```
Can you move the contents of the `try` into a function and call it from the `try`, so we don't add more nesting.

wwwouter commented:
I've put the code in an `getResponse` function and added a test.
This test will fail with the old code, but succeeds with the changes.

I've also added some files to work on VS Code on Windows. I can remove these if you want.

wwwouter commented:
Also: Travis is timing out, but this shows it's ok for at least node 4 and 8: https://travis-ci.org/sindresorhus/got/builds/346221646

kevva commented:
> I've also added some files to work on VS Code on Windows. I can remove these if you want.

Yes :).

wwwouter commented:
Ok, removed the .vscode folder!

sindresorhus commented:
Thanks :)

wwwouter commented:
You're welcome!
Could you take a quick peek at #461? 
Now at least I can catch the error, but it would be even better if there wasn't an error to catch :)
